crates/holochain/src/conductor/ribosome_store.rs

```diff
@@ -37,28 +37,40 @@ impl RibosomeStore {
 
     /// List all DNA hashes in the store.
     #[cfg_attr(feature = "instrument", tracing::instrument(skip(self)))]
-    pub fn list(&self) -> Vec<DnaHash> {
-        self.ribosomes.keys().cloned().collect()
+    pub fn list_dna_hashes(&self) -> Vec<DnaHash> {
+        self.ribosomes
+            .keys()
+            .cloned()
+            .map(|c| c.dna_hash().clone())
+            .collect()
     }
 
-    /// Get the DNA definition for a given DNA hash.
+    /// Get the DNA definition for a given CellId.
     #[cfg_attr(feature = "instrument", tracing::instrument(skip(self)))]
-    pub fn get_dna_def(&self, hash: &DnaHash) -> Option<DnaDef> {
+    pub fn get_dna_def(&self, cell_id: &CellId) -> Option<DnaDef> {
         self.ribosomes
-            .get(hash)
-            .map(|d| d.dna_def().clone().into_content())
+            .get(cell_id)
+            .map(|d| d.dna_def_hashed().clone().into_content())
     }
 
-    /// Get the DNA file for a given DNA hash.
+    /// Get the DNA file for a given CellId.
     // TODO: use Arc, eliminate cloning
     #[cfg_attr(feature = "instrument", tracing::instrument(skip(self)))]
-    pub fn get_dna_file(&self, hash: &DnaHash) -> Option<DnaFile> {
-        self.ribosomes.get(hash).map(|r| r.dna_file().clone())
+    pub fn get_dna_file(&self, cell_id: &CellId) -> Option<DnaFile> {
+        self.ribosomes.get(cell_id).map(|r| r.dna_file().clone())
+    }
+
+    /// Get the ribosome for a given CellId.
+    pub fn get_ribosome(&self, cell_id: &CellId) -> Option<RealRibosome> {
+        self.ribosomes.get(cell_id).cloned()
     }
 
-    /// Get the ribosome for a given DNA hash.
-    pub fn get_ribosome(&self, hash: &DnaHash) -> Option<RealRibosome> {
-        self.ribosomes.get(hash).cloned()
+    /// Get any ribosome associated to a CellId matching the given DnaHash.
```

Comment:
This is messy, but I see that it's fine for the time being while coordinator and integrity are still bunched together.